rust IP: Preserving existing IP setting

cathay4t opened the pull request:
If IP config not mentioned, we should preserving existing IP setting.

qinqon commented:
@cathay4t Can we create a test with https://github.com/nmstate/kubernetes-nmstate/pull/966#issuecomment-1018288282 ?

cathay4t committed:
rust IP: Preserving existing IP setting

If IP config not mentioned, we should preserving existing IP setting.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
Test case included.

cathay4t commented:
Got Ci failure. Checking.

cathay4t commented:
Included new patch to fix the problem exposed by this PR:
 * Metric of route should be used when comparing, or else, we cannot change metric of existing route.

cathay4t committed:
nm route: metric should be included when comparing

Problem:
    Cannot change metric of existing route

Root cause:
    Current code does not use metric when comparing, which cause desire
    route been dropped during de-duplication process.

Fix:
    Include metric for comparing and ordering.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>
